torch/_prims_common/wrappers.py

```diff
@@ -194,21 +194,34 @@ def _safe_copy_out(
 
 
 def out_wrapper(*out_names: str, exact_dtype: bool = False):
-    is_tensor = len(out_names) == 0
-    assert is_tensor or len(out_names) >= 2
+    # The wrapped function needs conversation of output parameters to ensure
+    # compatability between the Python API (which always uses "out" as the
+    # parameter name and may be a tuple) and the Aten API (which may have
+    # multiple output parematers and use different parameter names such as
+    # "grad_input", "indices" or "values".)
+
+    default_out_names = ("out",)
+    if len(out_names) == 0:
+        # Use default in out na,me
+        out_names = default_out_names
+
+    # If there is one tensor and the output name is "out", simply return the
+    # Tensor. Otherwise return it as as single named tuple to encode the name
+    # of the parameter.
+    is_tensor_named_out = (out_names == default_out_names)
 
     def _out_wrapper(fn: Callable) -> Callable:
         """
         Adds the out parameter to a Python reference.
         """
         out_type = (
             TensorLikeType
-            if is_tensor
+            if is_tensor_named_out
             else Tuple[tuple(TensorLikeType for _ in range(len(out_names)))]
         )
```

Comment:
This is not correct, as you are advertising in the signature that this function returns a tuple, which is not correct. We should signal in a different way to `register_decomposition` that it needs to overwrite the `out` param without changing so much the signature.